lib/publiccloud/basetest.pm

```diff
@@ -127,7 +127,16 @@ sub _cleanup {
 
 sub post_fail_hook {
     my ($self) = @_;
-    $self->_cleanup() unless $self->{cleanup_called};
+    unless ($self->{cleanup_called}) {
+        my $instance = $self->{run_args}->{my_instance};
+        if ($instance->ssh_script_run(cmd => 'sudo stat --printf="%s" /var/log/cloudregister', proceed_on_failure => 1) != 0) {
+            $instance->upload_log('/var/log/cloudregister', log_name => $autotest::current_test->{name} . '-cloudregister.log', failok => 1);
+        }
+        else {
+            bmwqemu::fctwarn("Test failed but cloudregister log is empty so skipping upload");
```

Comment:
What's `bmwqemu::fctwarn`?

Reply:
it is function which send a message into os-autoinst log , but looking for URL to explain you this better makes me realize that actually   `bmwqemu::fctwarn` is (deprecated)[https://github.com/os-autoinst/os-autoinst/blob/master/bmwqemu.pm#L63]   I need to use (log::fctwarn)[https://github.com/os-autoinst/os-autoinst/blob/master/log.pm#L55].

P.S.  this is perfect example why I hate perl ...